add docs and blog post how to switch from `master` to `main` (#3352)
nf-core/website: sites/main-site/src/content/blog/2025/switching-master-to-main.mdx

---
title: "nf-core goes woke!"
subtitle: "Announcing the possibility to change a pipeline default branch to main"
pubDate: 2025-05-15T12:00:00+01:00
headerImage: "https://images.unsplash.com/photo-1548710332-f65936b2ba27"
headerImageAlt: "branch with green leave"
embedHeaderImage: true
authors:
    - "mirpedrol"
    - "jfy133"
label:
    - "pipelines"
---

import { YouTube } from "@astro-community/astro-embed-youtube";

> # nf-core goes woke!

Reviewer: fixed some things, no worries. good for now

---
title: "nf-core goes woke!"
subtitle: "Announcing the possibility to change a pipeline default branch to main"

> pubDate: 2025-05-14T12:00:00+01:00

Reviewer: don't forget to update this before merging

Author: ```suggestion
pubDate: 2025-05-15T12:00:00+01:00
```

> headerImage: "https://unsplash.com/photos/macro-photography-of-green-plants-KzA5drJC-ok"

Reviewer: ```suggestion
headerImage: "https://images.unsplash.com/photo-1548710332-f65936b2ba27?q=80&w=2940&auto=format&fit=crop&ixlib=rb-4.1.0&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D"
```

headerImageAlt: "branch with green leave"
embedHeaderImage: true
authors:

>     - "jfy133"
>     - "mirpedrol"

Reviewer 2: **suggestion:** 
```suggestion
    - "mirpedrol"
     - "jfy133"
```
You're the main architect here!

label:
    - "pipelines"
---

> # nf-core goes woke!

Reviewer: ✊🏻

We are happy to announce that nf-core is rolling out full support for the use of `main` branches in nf-core pipeline GitHub repositories!

This brings us in-line with the rest of the GitHub programming community since the switch to the GitHub default back in [2020](https://github.blog/changelog/2020-10-01-the-default-branch-for-newly-created-repositories-is-now-main/).

> We recognise that changing the primary branch that is critical for pulling all pipelines can feel intimidating.

Reviewer 2: **suggestion:** 
```suggestion
We recognise that changing the primary branch that is critical for the pulling of any Nextflow pipelines can feel intimidating.
```

> We recognise that changing the primary branch that is critical for pulling all pipelines could feel intimidating.

Reviewer: ```suggestion
We recognise that changing the primary branch that is critical for pulling all pipelines can feel intimidating.
```

> So for Gen Z we have made a short video and for Millenials and older, a documentation page describing the steps to switch the default branch of existing pipelines:

Reviewer: 🔥

- Watch the video:

    <video
        src="assets/images/blog/switching-master-to-main/screen-capture.mp4"
        width="320"
        height="240"
        controls
    ></video>

- Read the [Documentation](https://nf-co.re/docs/tutorials/switching_branch_to_main/switching_master_to_main)

> For new pipelines, you can already select the primary branch on repository creation using `nf-core pipelines create`

Author: ```suggestion
For new pipelines, you can already select the primary branch on repository creation using `nf-core pipelines create`.
```

> If you have any problems, please shout on #tools channel on slack.

Reviewer: ```suggestion
If you have any problems, please shout on [#tools channel](https://nfcore.slack.com/archives/CE5LG7WMB) on slack.
```
